CHANGELOG.md

```diff
@@ -10,6 +10,12 @@
 
   The `statefile` parameter for the `resolve_reference` task has been replaced with a `state` parameter to maintain consistency among the other tasks and plans in the module.
 
+### Bug fixes
+
+* **Raise error when remote state cannot be loaded** ([#1436](https://github.com/puppetlabs/bolt/issues/1436))
+
+  When attempting to load remote state from a non-existent state file, Bolt would raise an obscure error. The `terraform` plugin now checks whether the attempt to load remote state returned any data and raises a more descriptive error if not.
```

Comment:
This is a bit misleading - the error was that loading a remote state file that didn't exist would return nil, which would be loaded into the inventory and cause an error. Pulling from a remote statefile that doesn't exist doesn't itself raise an error.